src/components/NavigatorSearch/modes/orders.ts

```diff
@@ -17,31 +17,30 @@ export function getGqlOrderId(orderNumber: string): string {
 }
 
 function getOrdersModeActions(
-  query: string,
   intl: IntlShape,
   navigate: UseNavigatorResult,
-  order: CheckIfOrderExistsQuery["order"],
+  orders: QuickOrderSearchResult,
 ): QuickSearchAction[] {
-  const gqlId = getGqlOrderId(query);
-
-  if (isQueryValidOrderNumber(query) && maybe(() => order.id === gqlId)) {
-    return [
-      {
-        extraInfo: transformOrderStatus(order.status, intl).localized,
-        label: intl.formatMessage(messages.goToOrder, {
-          orderNumber: query,
-        }),
-        onClick: () => {
-          navigate(orderUrl(gqlId));
-
-          return false;
-        },
-        type: "action",
-      },
-    ];
+  if (!orders) {
+    return [];
   }
 
-  return [];
+  return orders.map(order => {
+    const orderNumber = maybe(() => order.number, "");
+
+    return {
+      extraInfo: transformOrderStatus(order.status, intl).localized,
+      label: intl.formatMessage(messages.goToOrder, {
+        orderNumber,
+      }),
+      onClick: () => {
+        navigate(orderUrl(order.id));
+
+        return false;
+      },
+      type: "action",
+    };
```

Comment:
suggestion: you could extract this functio, name it like `toAction`, a use like this
```
return orders.map(toAction)
```